services/src/main/java/org/exoplatform/news/service/NewsTargetingService.java

```diff
@@ -33,11 +34,31 @@
    */
   List<NewsTargetingEntity> getTargets();
 
+  /**
+   * Gets news targets by id
+   * @param newsId
+   * @return {@link List} of news targets by id
+   */
+  List<String> getTargetsByNewsId(String newsId);
+
   /**
    * Gets  referenced targets
    *
    * @return {@link List} of referenced targets
    */
   List<NewsTargetingEntity> getReferencedTargets();
 
+
+  /**
+   * Set news targets
+   *@param news
```

Comment:
In terms of API definition, this is a very poor documentation. I had to read the implementation to know what behavior is used and the expected input + output of the method. please add a concrete and good documentation by thinking that it will serve for you in the future and especially for other developers that aims to use your API.